packages/ui/Media/Media.js

```diff
@@ -12,7 +14,69 @@ import {forward} from '@enact/core/handle';
 import {on, off} from '@enact/core/dispatcher';
 
 /**
- * Event forwarding map for all of the supported media events. See https://reactjs.org/docs/events.html#media-events
+ * Generates a key representing the source node or nodes provided
+ *
+ * ```
+ * getKeyFromSource('path/file.mp4'); // 'path/file.mp4'
+ * getKeyFromSource(
+ * 	<source src="path/file.mp4" type="video/mp4" />
+ * ); // 'path/file.mp4'
+ * getKeyFromSource([
+ * 	<source src="path/file.mp4" type="video/mp4" />,
+ * 	<source src="path/file.ogg" type="video/ogg" />,
+ * ]); // 'path/file.mp4+path/file.ogg'
+ * ```
+ *
+ * @param   {String|Element|Element[]} source URI for a source, `<source>` node, or array of
+ *                                     `<source>` nodes
+ * @returns {String}                   Key representing sources
+ * @function
+ * @memberof ui/Media
+ * @public
+ */
+const getKeyFromSource = (source = '') => {
+	if (React.isValidElement(source)) {
+		return React.Children.toArray(source)
+			.filter(s => !!s)
+			.map(s => s.props.src)
```

Comment:
I don't think the `type` will be relevant for determining uniqueness. For it to matter, two URIs would need to result in different physical sources. While this is possible, including the `type` wouldn't help us disambiguate them because (afaik) the `type` is only a hint to the browser and isn't sent to server hosting the video so the browser would get the same source for the same URI regardless of `type`.